includes/bp-messages/classes/class-bp-rest-sitewide-notices-endpoint.php

```
@@ -0,0 +1,917 @@
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$retval = array(
+			$this->prepare_response_for_collection(
+				$this->prepare_item_for_response( $notice, $request )
+			),
+		);
+
+		$response = rest_ensure_response( $retval );
+
+		/**
+		 * Fires after a sitewide notice is created via the REST API.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param BP_Messages_Notice $notice   Notice object.
+		 * @param WP_REST_Response   $response The response data.
+		 * @param WP_REST_Request    $request  The request sent to the API.
+		 */
+		do_action( 'bp_rest_sitewide_notices_create_item', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request has access to create a notice.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 */
+	public function create_item_permissions_check( $request ) {
+		$retval = $this->manage_item_permissions_check( $request );
+
+		/**
+		 * Filter the sitewide notices `create_item` permissions check.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_rest_sitewide_notices_create_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Update a notice.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function update_item( $request ) {
+		$test_notice = $this->get_notice_object( $request['id'] );
+		if ( ! $test_notice->id ) {
+			return new WP_Error(
+				'bp_rest_sitewide_notices_update_failed',
+				__( 'There was an error trying to update the notice.', 'buddypress' ),
+				array(
+					'status' => 500,
+				)
+			);
+		}
+
+		$notice = $this->prepare_item_for_database( $request );
+		$notice->save();
```

If it fails, it should return an error here.